docs/api.yaml

```diff
@@ -0,0 +1,641 @@
+        500:
+          $ref: "#/components/responses/500"
+
+  /courses/student:
+    post:
+      tags:
+        - "Course"
+      summary: "Enroll a student to a course"
+      description: "Create a new enrolled course for a student"
+      operationId: "enrollCourse"
+      requestBody:
+        description: "The student and the course to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/EnrolledCourse'
+      responses:
+        200:
+          description: "The created EnrolledCourse"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EnrolledCourse'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /courses/teacher:
+    post:
+      tags:
+        - "Course"
+      summary: "Attribute a course to a teacher"
+      description: "Create a new relation between a teacher and a teaching unit"
+      operationId: "attributeTeacherCourse"
+      requestBody:
+        description: "The teacher and the teaching unit to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TaughtCourse'
+      responses:
+        200:
+          description: "The created TaughtCourse"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TaughtCourse'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /activation:
+    patch:
+      tags:
+        - "Student"
+      summary: "Change activation status of a student"
+      description: ""
+      operationId: "changeActivation"
+      requestBody:
+        description: "The activation status of a student"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /impersionation:
```

Comment:
No. Impersonation is about calling some endpoints of the API while faking being someone else. Typically we want staff people to be able to call the GET endpoints (and only the GET endpoints as we want impersonation to be read-only) while pretending for example to be logged as a specific teacher or a specific student. I don't feel like we need to create a specific endpoint for doing this. Just proper definition of the STAFF role at implementation level.